packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+		);
+	} );
+
+	it( `shouldn't call any deps if only initialized`, () => {
+		createExPlatClient( createMockedConfig() );
+
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+
+	it( `shouldn't throw if initialized multiple times`, () => {
+		const a = createExPlatClient( createMockedConfig() );
+		const b = createExPlatClient( createMockedConfig() );
+		const c = createExPlatClient( createMockedConfig() );
+		const d = createExPlatClient( createMockedConfig() );
+		expect( a ).not.toBe( null );
+		expect( b ).not.toBe( null );
+		expect( c ).not.toBe( null );
+		expect( d ).not.toBe( null );
+	} );
+} );
+
+describe( 'ExPlatClient.loadExperimentAssignment single-use', () => {
+	it( `should successfully load an ExperimentAssignment`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig() );
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `[anonId] should successfully load an ExperimentAssignment`, async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( 'the-anon-id-123', ZERO_DELAY ) );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig() );
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `[developmentMode] should successfully load an ExperimentAssignment`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig( DEVELOPMENT_MODE ) );
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `Invalid experimentName: should return fallback and log`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig() );
+		await expect(
+			client.loadExperimentAssignment( 'the-invalid-experiment-name' )
+		).resolves.toEqual( {
+			experimentName: 'fallback_experiment_assignment',
```

Comment:
As ntoed on #49829, I think there's no harm in returning the invalid name here. It's less surprising than `fallback_experiment_assignment`.

Reply:
See the comment there - it is not a valid ExperimentAssignment and would fail various run-time checks.

Reply:
I have replied there and have also changed the name to indicate that it is an invalid name.